### PR TITLE
Install Pyccel from GitHub for devel tests

### PR DESCRIPTION
**Pull request by yguclu**
If we install Pyccel from sources after cloning the Git repository, the command `pip freeze` will return a local file path for the `pyccel` package, and hence it cannot be used to reproduce the same environment on another machine (see [this comment](https://github.com/pyccel/pyccel-benchmarks/pull/9#discussion_r1046212334)). We avoid this problem by passing the correct repository address directly to the `pip install` command.

### CODE REVIEW

**Review by EmilyBourne — APPROVED** [review]

**Review by EmilyBourne — APPROVED** [review]

File: version_specific_results/devel_performance_310_requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ numpy==1.23.5
 packaging==22.0
 Pillow==9.3.0
 ply==3.11
-pyccel @ file:///home/runner/work/pyccel-benchmarks/pyccel-benchmarks/pyccel
+pyccel @ git+https://github.com/pyccel/pyccel@4e703de8b2a95eac64fb8971269d987c5ba36f49
 pyccel-benchmarks @ file:///home/runner/work/pyccel-benchmarks/pyccel-benchmarks
```
Comment (aholmes) [inline version_specific_results/devel_performance_310_requirements.txt:19]: Just checking - should this one be updated too?

Comment (yguclu) [reply version_specific_results/devel_performance_310_requirements.txt:19]: Absolutely! 😅

Comment (EmilyBourne) [reply version_specific_results/devel_performance_310_requirements.txt:19]: I'm not sure why this only appears in this file though!

Comment (yguclu) [reply version_specific_results/devel_performance_310_requirements.txt:19]: I'll take care of this

**Pull request merged**